src/Controllers/DatabaseController.js

```diff
@@ -220,8 +220,8 @@ const filterSensitiveData = (
     protectedFields && protectedFields.forEach(k => delete object[k]);
 
     // fields not requested by client (excluded),
-    //but were needed to apply protecttedFields
-    perms.protectedFields &&
+    // but were needed to apply protectedFields
+    perms && perms.protectedFields &&
       perms.protectedFields.temporaryKeys &&
       perms.protectedFields.temporaryKeys.forEach(k => delete object[k]);
```

Comment:
Since we're on Node >=14 now we can finally use optional chaining.

```suggestion
    perms?.protectedFields?.temporaryKeys?.forEach(k => delete object[k]);
```

Reply:
First use of optional chaining in the entire repo

Reply:
Deserves a cake!